src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp

```diff
@@ -209,24 +195,21 @@ void SyncInferRequest::set_tensors_impl(const ov::Output<const ov::Node> port, c
 
     OPENVINO_ASSERT(is_host || is_remote, "[GPU] Incorrect input blobs. All blobs must be of the same type");
 
-    for (const auto& input : get_inputs()) {
-        if (input == port) {
-            m_batched_tensors[input.get_tensor_ptr()] = tensors;
-            return;
-        }
-    }
-    OPENVINO_THROW("[GPU] Cannot find input tensors for port ", port);
+    size_t port_index = find_port(port).idx;
+    OPENVINO_ASSERT(m_input_ports_map.count(port_index) != 0, "[GPU] Cannot find input tensors for port ", port, " with index ", port_index);
+    const auto& tensor = m_input_ports_map.at(port_index).get_tensor_ptr();
+    m_batched_tensors[tensor] = tensors;
 }
 
 ov::SoPtr<ov::ITensor> SyncInferRequest::get_tensor(const ov::Output<const ov::Node>& port) const {
     bool is_input = ov::op::util::is_parameter(port.get_node());
-    const auto name = get_port_name(port);
+    size_t port_index = find_port(port).idx;
```

Comment:
let's drop usage of `ov::op::util::is_parameter` and rely on port_type returned by `find_port`